### ISSUE TITLE
sipi unauthorized response is not processed (salsah 1.5)

### ISSUE THREAD

**Issue by loicjaouen** [dhlab-basel/Knora#487]
in Salsah 1.5, when creating a resource with a file uploaded to sipi, after the creation, when viewing the resource, if sipi returns a `non authorized` response, it is not processed by Salsah 1.5 and ends up in a time-out.

![capture d ecran 2017-04-18 a 10 23 30](https://cloud.githubusercontent.com/assets/3933654/25121900/7ba9f62c-2423-11e7-98e4-d1d3a3e6f1b1.png)
![capture d ecran 2017-04-18 a 10 23 44](https://cloud.githubusercontent.com/assets/3933654/25121905/7dcbe28a-2423-11e7-9271-08ecc372ec19.png)

**Comment by tobiasschweizer** [comment]
The question is why you would get no permissions to view the image when you are the owner of the resource.

Could you please try to run the integration tests and see if they pass: `it:test` in sbt?

**Comment by loicjaouen** [comment]
> The question is why you would get no permissions to view the image when you are the owner of the resource.

no, this is not the question :)

Tests are done to catch things we do that are just plain stupid or that look fine but have subtle side effects.  
The first time there was a problem in permissions definition, the second time, to replicate it, I pointed sipi to a different knora instance (or non existing knora maybe).

Anyway; the causes are many and unpredictable.

The question for this issue is therefore not the cause but how to catch the unexpected response without waiting for the timeout.

**Comment by tobiasschweizer** [comment]
But have you checked if Sipi is able to parse the Knora session id out of the cookie sent in the header?

**Comment by loicjaouen** [comment]
yes, the test is fine ;) I had to abuse the system when you asked me to reproduce the problem because: the original problem was fixed and its root was not a faulty test.

**Comment by benjamingeer** [comment]
Please provide precise steps to reproduce this issue, along with all necessary test data, in the format given here:

https://github.com/dhlab-basel/Knora/issues/492#issuecomment-298628691
